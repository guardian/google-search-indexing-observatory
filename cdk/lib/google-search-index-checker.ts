import { join } from "path";
import type { GuStackProps } from "@guardian/cdk/lib/constructs/core";
import { GuStack } from "@guardian/cdk/lib/constructs/core";
import type { App } from "aws-cdk-lib";
import { CfnInclude } from "aws-cdk-lib/cloudformation-include";
import {GuScheduledLambda} from "@guardian/cdk";
import {Runtime} from "aws-cdk-lib/aws-lambda";
import {Schedule} from "aws-cdk-lib/aws-events";
import {Duration} from "aws-cdk-lib";
import {NoMonitoring} from "@guardian/cdk/lib/constructs/cloudwatch";

export class GoogleSearchIndexChecker extends GuStack {
  constructor(scope: App, id: string, props: GuStackProps) {
    super(scope, id, props);
    const noMonitoring: NoMonitoring = { noMonitoring: true };
    new GuScheduledLambda(this, "scheduledLambda", {
      app: "google-search-index-checker",
      fileName: "google-search-index-checker.jar",
      handler: "ophan.google.index.checker.Lambda::handler",
      monitoringConfiguration: noMonitoring,
      rules: [{ schedule: Schedule.rate(Duration.minutes(1)) }],
      runtime: Runtime.JAVA_11
    })
  }
}
