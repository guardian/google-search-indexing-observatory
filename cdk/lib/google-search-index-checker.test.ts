import { App } from "aws-cdk-lib";
import { Template } from "aws-cdk-lib/assertions";
import { GoogleSearchIndexChecker } from "./google-search-index-checker";

describe("The GoogleSearchIndexChecker stack", () => {
  it("matches the snapshot", () => {
    const app = new App();
    const stack = new GoogleSearchIndexChecker(app, "GoogleSearchIndexChecker", { stack: "ophan", stage: "TEST" });
    const template = Template.fromStack(stack);
    expect(template.toJSON()).toMatchSnapshot();
  });
});
