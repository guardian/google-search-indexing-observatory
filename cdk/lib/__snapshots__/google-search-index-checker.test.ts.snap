// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The GoogleSearchIndexChecker stack matches the snapshot 1`] = `
Object {
  "Metadata": Object {
    "gu:cdk:constructs": Array [],
    "gu:cdk:version": "TEST",
  },
}
`;
